core/server/common/src/main/java/alluxio/master/AbstractMaster.java

```diff
@@ -313,4 +323,103 @@ public void close() {
       }
     }
   }
+
+  @NotThreadSafe
+  public final class MasterAuditContext implements AuditContext {
+    private final AsyncUserAccessAuditLogWriter mAsyncAuditLogWriter;
+    private boolean mAllowed;
+    private boolean mCommitted;
+    private String mCommand;
+    private String mSrcPath;
+    private String mDstPath;
+    private String mUser;
+    private String mIp;
+    private String mSrcPathOwner;
+    private String mSrcPathGroup;
+    private short mSrcPathMode;
+
+    @Override
+    public MasterAuditContext setAllowed(boolean allowed) {
+      mAllowed = allowed;
+      return this;
+    }
+
+    @Override
+    public void setCommitted(boolean committed) { mCommitted = committed; }
+
+    public MasterAuditContext setCommand(String command) {
+      mCommand = command;
+      return this;
+    }
+
+    public MasterAuditContext setSrcPath(String srcPath) {
+      mSrcPath = srcPath;
+      return this;
+    }
+
+    public MasterAuditContext setDstPath(String dstPath) {
+      mDstPath = dstPath;
+      return this;
+    }
+
+    public MasterAuditContext setUser(String user) {
+      mUser = user;
+      return this;
+    }
+
+    public MasterAuditContext setIp(String ip) {
+      mIp = ip;
+      return this;
+    }
+
+    public MasterAuditContext setSrcPathOwner(String owner) {
+      mSrcPathOwner = owner;
+      return this;
+    }
+
+    public MasterAuditContext setSrcPathGroup(String group) {
+      mSrcPathGroup = group;
+      return this;
+    }
+
+    public MasterAuditContext setSrcPathMode(short mode) {
+      mSrcPathMode = mode;
+      return this;
+    }
+
+    private MasterAuditContext(AsyncUserAccessAuditLogWriter asyncAuditLogWriter) {
+      mAsyncAuditLogWriter = asyncAuditLogWriter;
+      mAllowed = true;
+    }
+
+    @Override
+    public void append() {
+      if (mAsyncAuditLogWriter != null) {
+        mAsyncAuditLogWriter.append(this);
+      }
+    }
+
+    @Override
+    public boolean isCommitted() { return mCommitted; }
+
+    @Override
+    public void close() {
+      if (mAsyncAuditLogWriter != null) {
+        mAsyncAuditLogWriter.commit(this);
+      }
+    }
+
+    @Override
+    public String toString() {
+      if (mAllowed) {
+        return String.format("allowed=true\tuser=%s\tip=%s\tcmd=%s\tsrc=%s\tdst=%s\tperm=%s:%s:%d",
+            mUser, mIp, mCommand, mSrcPath, mDstPath, mSrcPathOwner, mSrcPathGroup, mSrcPathMode);
+      } else {
+        return String.format("allowed=false\tuser=%s\tip=%s\tcmd=%s\tsrc=%s\tdst=%s\tperm=null",
+            mUser, mIp, mCommand, mSrcPath, mDstPath);
+      }
+    }
+  }
+
+  protected AuditContext createAuditContext() { return new MasterAuditContext(mAsyncAuditLogWriter); }
```

Comment:
I think AuditContext is similar to JournalContext.
It does not have to be static, because we will calling this method in non-static methods of DefaultFileSystemMaster.
I am OK with either, but the createJournalContext method return type is JournalContext, therefore, I guess there may be some thoughts put into the design of the method signature. It will be good if I can know more about this.